src/main/java/be/appwise/core/networking/proxyman/Message.kt

```diff
@@ -0,0 +1,49 @@
+package be.appwise.core.networking.proxyman
+
+import android.os.Build
+import android.provider.Settings
+
+
+/**
+ * if requestbody is filled in succes ?
+ * */
+class Message(
+    val id: String,
+    val messageType: String,
+    val buildVersion: String = "0.9.0",
+    var content: Data?
+) {
+
+    class Project {
+        companion object {
+            val current = Project()
+        }
+
+        val name: String = ProxyManNetworkDiscoveryManager.getAppContext().applicationInfo.loadLabel(ProxyManNetworkDiscoveryManager.getAppContext().packageManager).toString()
+        val bundleIdentifier: String = ProxyManNetworkDiscoveryManager.getAppContext().packageName
+    }
+
+    class Device {
+        companion object {
+            val current = Device()
+        }
+
+        var name: String =
+            Settings.Secure.getString(ProxyManNetworkDiscoveryManager.getAppContext().contentResolver, "bluetooth_name")
```

Comment:
Is this the correct name? I have been testing device name myself in the last week and I found this field https://developer.android.com/reference/android/provider/Settings.Global#DEVICE_NAME

Reply:
This is the same field!! Go on your device to settings > about phone > device name. It mentions this is the name that is used for wifi and bluetooth. When I changed it in the bluetooth and wifi settings it also changed on this screen.

Reply:
@dieter115 Not really 😅 I went through AOSP during my investigation
And changing device name in settings will indeed also change wifi hotspot and bluetooth names. But changing bluetooth naam will not change device name. So the device name and bluetooth name can be different.
Setting device name: https://github.com/aosp-mirror/platform_packages_apps_settings/blob/c5a500bf07f33e02ff3d315d0ceddf9c2d31d000/src/com/android/settings/deviceinfo/DeviceNamePreferenceController.java#L127
Setting bluetooth name: https://github.com/aosp-mirror/platform_packages_apps_settings/blob/c5a500bf07f33e02ff3d315d0ceddf9c2d31d000/src/com/android/settings/bluetooth/LocalDeviceNameDialogFragment.java#L87

Reply:
On my device he changes the device name when i change the bluetooth name and also in the other direction.

Reply:
Maybe it has something to do with Manufacturer specific things? Maybe Pixel devices will have the changes propagated both ways, while i.e. Samsung does something different? 🤔

Reply:
Can be :) . i'll rework it ! Thanks for testing it !

Reply:
@Glennmen i tested this and your fields returns oneplus 8 pro in my case. Not the device name you set in the settings. This will be a problem for all the people who have the oneplus 8 pro...